migration/blocktransactions/blocktransactions.go

```diff
@@ -0,0 +1,291 @@
+package blocktransactions
+
+import (
+	"context"
+	"errors"
+	"iter"
+	"runtime"
+	"time"
+
+	"github.com/NethermindEth/juno/core"
+	"github.com/NethermindEth/juno/db"
+	"github.com/NethermindEth/juno/db/typed/key"
+	"github.com/NethermindEth/juno/db/typed/prefix"
+	"github.com/NethermindEth/juno/encoder"
+	_ "github.com/NethermindEth/juno/encoder/registry"
+	"github.com/NethermindEth/juno/utils"
+	"github.com/fxamacker/cbor/v2"
+	"github.com/sourcegraph/conc/stream"
+)
+
+const (
+	batchSize        = 100
+	batchConcurrency = 4
+	timeLogRate      = 5 * time.Second
+)
+
+type BlockTransactionsMigration struct{}
+
+type blockTransactions struct {
+	Transactions []cbor.RawMessage `cbor:"1,keyasint,omitempty"`
+	Receipts     []cbor.RawMessage `cbor:"2,keyasint,omitempty"`
+}
+
+var (
+	rawTransactions = prefix.NewPrefixedBucket(
+		core.TransactionsByBlockNumberAndIndexBucket.RawKey().RawValue(),
+		prefix.Prefix(key.Uint64, prefix.Prefix(key.Uint64, prefix.End[[]byte]())),
+	)
+	rawReceipts = prefix.NewPrefixedBucket(
+		core.ReceiptsByBlockNumberAndIndexBucket.RawKey().RawValue(),
+		prefix.Prefix(key.Uint64, prefix.Prefix(key.Uint64, prefix.End[[]byte]())),
+	)
+	rawBlockTransactions = core.BlockTransactionsBucket.RawValue()
+)
+
+func collect(iter iter.Seq2[prefix.Entry[[]byte], error]) ([]cbor.RawMessage, error) {
+	items := []cbor.RawMessage{}
+	for item, err := range iter {
+		if err != nil {
+			return nil, err
+		}
+		items = append(items, item.Value)
+	}
+	return items, nil
+}
+
+func readBlock(database db.KeyValueStore, blockNumber uint64) (blockTransactions, bool, error) {
+	transactions, err := collect(rawTransactions.Prefix().Add(blockNumber).Scan(database))
+	if err != nil {
+		return blockTransactions{}, false, err
+	}
+
+	receipts, err := collect(rawReceipts.Prefix().Add(blockNumber).Scan(database))
+	if err != nil {
+		return blockTransactions{}, false, err
+	}
+
+	if len(transactions) == 0 || len(receipts) == 0 {
+		has, err := core.BlockTransactionsBucket.Has(database, blockNumber)
+		if err != nil {
+			return blockTransactions{}, false, err
+		}
+		if has {
+			return blockTransactions{}, false, nil
+		}
+	}
+
+	return blockTransactions{
+		Transactions: transactions,
+		Receipts:     receipts,
+	}, true, nil
```

Comment:
small nitpick: I believe the boolean here is redundant since `blockTransaction{}` is equivalent to  `false`.